src/main/java/org/elasql/util/TransactionFeaturesRecorder.java

```diff
@@ -0,0 +1,192 @@
+package org.elasql.util;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.vanilladb.core.server.VanillaDb;
+import org.vanilladb.core.server.task.Task;
+
+public class TransactionFeaturesRecorder extends Task {
```

Comment:
(Optional) Merge this class with `TransactionStatisticsRecorder` to avoid redundant code.